tests/e2e/tests/test_accommodation_v2.go

```
@@ -0,0 +1,466 @@
+	require.NoError(t, tt.caminoNetwork.Client.RegisterCMService(ctx, botGenerated.MintServiceV2))
+
+	supplierPartnerPlugin := tt.CreatePartnerPlugin(ctx, t)
+
+	supplierServices := []bot.CMService{
+		{Name: botGenerated.AccommodationProductListServiceV2, Fee: 100},
+		{Name: botGenerated.AccommodationProductInfoServiceV2, Fee: 110},
+		{Name: botGenerated.AccommodationSearchServiceV2, Fee: 120},
+		{Name: botGenerated.ValidationServiceV2, Fee: 130},
+		{Name: botGenerated.MintServiceV2, Fee: 140},
+	}
+
+	// bot with partnerPlugin and without rpc server (supplier)
+	supplierBot := tt.CreateBot(ctx, t, false, supplierPartnerPlugin, supplierServices)
+
+	// bot without partnerPlugin and with rpc server (distributor)
+	distributorBot := tt.CreateBot(ctx, t, true, nil, nil)
+
+	return supplierPartnerPlugin, supplierBot, distributorBot
+}
+
+/* Simple product list request which shall return all properties. Checking if one is the right one */
+func TestAccommodationProductListServiceV2(t *testing.T, tt *Test, distributorBot *bot.Bot, supplierBot *bot.Bot, ctx context.Context) {
+	hotelCode := "HOTEL123456"
+
+	resp, err := distributorBot.AccommodationProductListServiceV2.AccommodationProductList(
+		requestContext(ctx, &metadata.Metadata{
+			Recipient: supplierBot.CMAccountAddress().Hex(),
+		}),
+		&accommodationv2.AccommodationProductListRequest{
+			Header: &typesv1.RequestHeader{BaseHeader: &typesv1.Header{}},
+		},
+	)
+	require.NoError(t, err)
+
+	tt.logger.Debug("AccommodationProductListServiceV2.AccommodationProductList response:\n", protoMessageToJSON(tt, resp))
+
+	require.Equal(t, typesv1.StatusType_STATUS_TYPE_SUCCESS, resp.Header.Status, "unexpected response status")
+	require.Empty(t, resp.Header.Alerts, "unexpected response alerts")
+
+	// The response should contain all 5 properties defined by the pp-mock
+	require.Len(t, resp.Properties, 5, "unexpected number of properties in response")
```

would be nice if we can reference "5" from some pp-mock variable (e.g. const or array len or something)

---

Unfortunately not that easy as the underlying data is a json file in the context of the pp-mock and the result number is a result of filtering (in this case not filtering).
And in this particular case it's even better to have it decoupled as when we try to extract that from the pp-mock the implementation could be wrong leading to a false-positive test result.
This means that in case we change something in the pp-mock data an error will pop up which is IMHO fine as you need to be aware what's happening when changing stuff.
But I agree that it would be better to have it as a constant for better readability. Will change that.